### PR TITLE
Add alsa support.

### PR DESCRIPTION
**Pull request by gpgabriel**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1,37 +1,41 @@
...
+  // run process
+  if (os.platform() === 'darwin') {
+      ps = spawn(
+        opts.soxPath,
+        [ '--buffer', opts.buffer, '--bits', opts.bits, '--channels', opts.channels,
+          '--encoding', opts.encoding, '--rate', opts.rate, opts.inFile, '-p'
+        ]
+      )
+  } else {
+      ps = spawn(
```
Comment (ahdinosaur) [inline main.js]: what if instead of `os.platform() !== 'darwin'`, we check that `arecord` path exists to run this code path?

File: main.js [hunk 2]
```diff
@@ -1,37 +1,41 @@
...
-      opts.inFile,
-      '-p'
-    ]
-  )
+
+  // run process
+  if (os.platform() === 'darwin') {
+      ps = spawn(
+        opts.soxPath,
+        [ '--buffer', opts.buffer, '--bits', opts.bits, '--channels', opts.channels,
```
Comment (ahdinosaur) [inline main.js]: unnecessary formatting change

File: main.js [hunk 3]
```diff
@@ -1,37 +1,41 @@
...
+var rangeFit = require('range-fit');
+var intmin = require('compute-intmin');
+var intmax = require('compute-intmax');
+var pull = require('pull-stream');
+var toPull = require('stream-to-pull-stream');
 
 module.exports = readAudio
 
 function readAudio (opts, onAbort) {
+  var ps;
```
Comment (ahdinosaur) [inline main.js]: you can do `var ps =` in both code paths, no problem. or this is fine too.

File: main.js [hunk 4]
```diff
@@ -1,37 +1,41 @@
...
+var spawn = require('child_process').spawn;
+var defined = require('defined');
+var Ndsamples = require('ndsamples');
+var getDataType = require('dtype');
+var bufferToTypedArray = require('buffer-to-typed-array');
+var rangeFit = require('range-fit');
+var intmin = require('compute-intmin');
+var intmax = require('compute-intmax');
+var pull = require('pull-stream');
+var toPull = require('stream-to-pull-stream');
```
Comment (ahdinosaur) [inline main.js]: unnecessary changes to add all the semicolons

**Review by ahdinosaur — CHANGES_REQUESTED** [review]
hey @gpgabriel, thanks for this contribution. :smile_cat: 

keen to support `alsa`, but first gave some feedback.

cheers!

**Comment by gpgabriel** [comment]
@ahdinosaur Thanks for the detailed review. Let me know if there's anything else that you think could be improved.

File: main.js [hunk 5]
```diff
@@ -1,5 +1,7 @@
 'use strict';
 
+var os = require('os')
```
Comment (ahdinosaur) [inline main.js]: no longer used

**Review by ahdinosaur — APPROVED** [review]
thanks for incorporating the feedback @gpgabriel, looks great meow.

**Comment by ahdinosaur** [comment]
published as [`8.1.0`](https://github.com/ahdinosaur/read-audio/commit/fe28262866c9b6c29a0ea9744d40a72dd983e5fc)

**Pull request merged**